File: app/src/main/java/me/imzack/app/end/injector/component/TypeDetailComponent.java
package me.imzack.app.end.injector.component;

import me.imzack.app.end.injector.module.TypeDetailPresenterModule;
import me.imzack.app.end.injector.scope.ActivityScope;
import me.imzack.app.end.view.activity.TypeDetailActivity;

import dagger.Component;

@ActivityScope
@Component(dependencies = AppComponent.class, modules = TypeDetailPresenterModule.class)
public interface TypeDetailComponent {

    void inject(TypeDetailActivity typeDetailActivity);
}

File: app/src/main/kotlin/me/imzack/app/end/injector/component/TypeDetailComponent.kt
package me.imzack.app.end.injector.component

import dagger.Component
import me.imzack.app.end.injector.module.TypeDetailPresenterModule
import me.imzack.app.end.injector.scope.ActivityScope
import me.imzack.app.end.view.activity.TypeDetailActivity

@ActivityScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(TypeDetailPresenterModule::class))
interface TypeDetailComponent {

    fun inject(typeDetailActivity: TypeDetailActivity)
}
